### PR TITLE
First pass of new submission docs

### PR DESCRIPTION
**Pull request by mike-ferguson**
This PR moved the developer.md file into a new home, submission.rst. 

PR #418 will be merged into this when done 

This PR will be merged into #407 after #418 is folded in.

### CODE REVIEW

File: docs/source/modules/submission.rst [hunk 1]
```diff
@@ -1,12 +1,119 @@
...
-    :undoc-members:
-.. _model-tools: https://github.com/brain-score/model-tools
-.. _model-interface: https://github.com/brain-score/brain-score/blob/master/brainscore/model_interface.py
+Submission system Components
+#################
+
+
+
+To provide an automatic scoring mechanism for artificial models of the brain, Brain-Score has implemented a entire
+ecosystem, with each part explained below:
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:10]:
```suggestion
To provide an automatic scoring mechanism for artificial models of the brain, 
Brain-Score provides an automated submission system.
This enables us to automatically score new models on existing benchmarks,
as well as new benchmarks against existing models.
```

File: docs/source/modules/submission.rst [hunk 2]
```diff
@@ -1,12 +1,119 @@
...
+
+.. image:: docs/source/modules/brainscore_submission.png
+    :width: 200px
+    :align: center
+    :height: 100px
+    :alt: alternate text
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:16]:
```suggestion
    :alt: submission system diagram
```

File: docs/source/modules/submission.rst [hunk 3]
```diff
@@ -1,12 +1,119 @@
...
+
+- **Brain-Score Website**:
+
+    The `website <www.brain-score.org>`_ implementation, also found on
+    `Github <https://github.com/brain-score/brain-score.web>`_, handles our front end. It is implemented using Django
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:21]:
```suggestion
    The `website <www.brain-score.org>`_ (`Github <https://github.com/brain-score/brain-score.web>`_) handles our front end. It is implemented using Django
```

File: docs/source/modules/submission.rst [hunk 4]
```diff
@@ -1,12 +1,119 @@
...
+    and also accesses the database instance. When user submit their models in form of a zip file, the website triggers
+    the execution with an http request to Jenkins (see below), also delivering the submitted zip file. Users can also
+    see their profile, personal (private) leaderboard as well as the public leaderboard.
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:24]:
```suggestion
    and also accesses the database instance. 
    
    Users can submit plugins as a zip file via the website or as a pull request via Github. 
    Zip file submissions are converted into a pull request that is automatically merged if tests pass.
    Once a pull request has been merged and if there are new model or benchmark plugins, 
    a Github action triggers the scoring runs which are handled by Jenkins 
    and run on the OpenMind compute cluster.
```

File: docs/source/modules/submission.rst [hunk 5]
```diff
@@ -1,12 +1,119 @@
...
+ - **Jenkins**:
+
+    `Jenkins <http://braintree.mit.edu:8080/>`_ is a continuous integration tool, which we use to automatically run
+    project unittests and the scoring process for models of the brain. Jenkins is running on Braintree - DiCarlo lab's
+    internal server. Jenkins defines different jobs and executes different tasks. The task for a new submission is
+    triggered via the website and the unittest tasks are triggerd by GitHub web hooks. Once the jobs are triggered,
+    Jenkins runs a procedure to execute the tests or scoring and communicate the results back to the user (via Email)
+    or back to GitHub.
+
+-  **Openmind**
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:47]:
```suggestion
-  **OpenMind**
```

File: docs/source/modules/submission.rst [hunk 6]
```diff
@@ -1,12 +1,119 @@
...
+
+    As scoring submissions is a computationally and memory expensive process, we cannot execute model scoring on small
+    machines. We submit jobs to Openmind, a computer cluster operated by MIT BCS. The big advantage of Openmind is its
+    queuing system, which allows to define detailed resource requirements. Jobs are executed once their requested
+    resources are available. The Jenkins related contents are stored on ``/om5/group/dicarlo/jenkins``. This directory
+    contains a script for model submission (`score_model.sh`) and for unittests (`unittests_brainscore.sh`). The scripts
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:53]: @kvfairchild are these two scripts still accurate?

File: docs/source/modules/submission.rst [hunk 7]
```diff
@@ -1,12 +1,119 @@
...
+    are executed in an Openmind job and are responsible for fully installing a conda environment, executing the process,
+    and shutting everything down again. Results are stored in the database or copied to amazon S3 cloud file system.
+    From there, Jenkins reports the results back to its caller (the website) and emails the user.
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:56]:
```suggestion
    and shutting everything down again. 
    For scoring runs, results are stored in the database and sent to the user via email.
    For unit tests, results are reported back to Github.
```

File: docs/source/modules/submission.rst [hunk 8]
```diff
@@ -1,12 +1,119 @@
...
+
+
+What to do
+#################
+
+
+...when changing the database schema
+************************************
+The current schema is depicted `here
+<https://github.com/brain-score/brain-score/blob/master/brainscore/submission/db_schema.uml>`_:
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:86]: move this file into `docs/source/modules`

File: docs/source/modules/submission.rst [hunk 9]
```diff
@@ -1,12 +1,119 @@
...
+
+
+When the database schema has to be changed, use the `Brain-Score.web <https://github.com/brain-score/brain-score.web>`_
+project, along with django commands, to adjust the tables (in `benchmarl/models.py`). Once changes are made locally,
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:90]:
```suggestion
project, along with django commands, to adjust the tables (in `benchmark/models.py`). The schema also has to be updated in `core <https://github.com/brain-score/core/blob/main/brainscore_core/submission/database_models.py>`_. Once changes are made locally,
```

File: docs/source/modules/submission.rst [hunk 10]
```diff
@@ -1,12 +1,119 @@
...
+see `here < https://github.com/brain-score/brain-score.web/blob/master/deployment.md#to-deploy>`_ to apply those
+migrations to the correct databases. All needed changes to the database (dev or prod) should be done with Django via
+migrations. During development, work with the dev database (secret `brainscore-1-ohio-cred`); when your
+tests pass on the test database (`brainscore-ohio-test`) they are ready for the PR. Once the PR is approved and test
+cases run, the PR can be merged. Finally, apply those migrations to the prod database via the link above.
+
+
+...changing the submission process
+**********************************
+In addition to the main job for scoring submission (run_benchmarks), Jenkins contains a second job (dev_run_benchmarks),
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:100]:
```suggestion
In addition to the main job for scoring submission (`score_plugins`), Jenkins contains a second job (`dev_score_plugins`),
```
@kvfairchild correct?

File: docs/source/modules/submission.rst [hunk 11]
```diff
@@ -1,12 +1,119 @@
...
+which can be used to test new submission code. It is also a good idea instead of checking out the Brain-Score master
+branch, as it is done in the default job, to checkout your development branch instead. This way you can run a whole
+submission without harming the "production" job. This is accomplished already by a duplicate of score_models.sh for dev,
+aptly named dev_score_models.sh. That script is what is run on Jenkins's dev environment. Once the development job runs
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:104]: @kvfairchild still the correct script names?

File: docs/source/modules/submission.rst [hunk 12]
```diff
@@ -1,12 +1,119 @@
...
+successfully, the code can be merged to master and will be run "in production".
+
+Scoring Process Description
+#################
+For scoring submitted files, we install the Brain-Score framework on Opemnind and run the scoring process. There are
+two types of submissions possible:
```
Comment (mschrimpf) [inline docs/source/modules/submission.rst:110]: @kvfairchild not sure this is still the best way to describe

**Review by mschrimpf — CHANGES_REQUESTED** [review]

File: docs/source/modules/submission.rst [hunk 13]
```diff
@@ -1,12 +1,119 @@
...
+    :width: 200px
+    :align: center
+    :height: 100px
+    :alt: alternate text
+
+- **Brain-Score Website**:
+
+    The `website <www.brain-score.org>`_ implementation, also found on
+    `Github <https://github.com/brain-score/brain-score.web>`_, handles our front end. It is implemented using Django
+    and also accesses the database instance. When user submit their models in form of a zip file, the website triggers
```
Comment (kvfairchild) [inline docs/source/modules/submission.rst]:
```suggestion
    and also accesses the database instance. When a user submits their models in form of a zip file, the website triggers
```

**Pull request merged**